lib/livebook/live_markdown/import.ex

```diff
@@ -527,16 +527,28 @@ defmodule Livebook.LiveMarkdown.Import do
   defp postprocess_stamp(notebook, _notebook_source, nil, _), do: {notebook, []}
 
   defp postprocess_stamp(notebook, notebook_source, stamp_data, stamp_hub_id) do
-    hub = Hubs.get_offline_hub(stamp_hub_id) || Hubs.fetch_hub!(stamp_hub_id)
+    {hub, offline?} =
+      cond do
+        hub = Hubs.get_offline_hub(stamp_hub_id) -> {hub, true}
+        hub = Hubs.fetch_hub!(stamp_hub_id) -> {hub, false}
+      end
 
-    with %{"offset" => offset, "stamp" => stamp} <- stamp_data,
-         {:ok, notebook_source} <- safe_binary_slice(notebook_source, 0, offset),
-         {:ok, metadata} <- Livebook.Hubs.verify_notebook_stamp(hub, notebook_source, stamp) do
-      notebook = apply_stamp_metadata(notebook, metadata)
-      {notebook, []}
-    else
-      _ -> {notebook, ["failed to verify notebook stamp"]}
-    end
+    {valid_stamp?, notebook, messages} =
+      with %{"offset" => offset, "stamp" => stamp} <- stamp_data,
+           {:ok, notebook_source} <- safe_binary_slice(notebook_source, 0, offset),
+           {:ok, metadata} <- Livebook.Hubs.verify_notebook_stamp(hub, notebook_source, stamp) do
+        notebook = apply_stamp_metadata(notebook, metadata)
+        {true, notebook, []}
+      else
+        _ -> {false, notebook, ["failed to verify notebook stamp"]}
+      end
+
+    # We enable teams features for offline hub only if the stamp
+    # is valid, which ensures it is an existing Teams hub
+    teams_enabled = is_struct(hub, Livebook.Hubs.Team) and (not offline? or valid_stamp?)
```

Comment:
Could we instead have this information be returned as part of the stamp metadata and applied in `apply_stamp_metadata`?

Reply:
When the hub is online and exists we want to have `teams_enabled: true`, even if the stamp is missing or invalid.

Other than that I think stamp/verify should be symmetric, we shouldn't add more keys to the metadata that were not stored there originally.

Reply:
Ok. I am not a big fan of checking for the struct iself but this is fine for now. :)